src/app/detail/ui/ToggleController.tsx

```diff
@@ -0,0 +1,29 @@
+'use client'
+
+import { selectTaskDetails, setShowSidebar } from '@/redux/features/taskDetailsSlice'
+import store from '@/redux/store'
+import { Box, useMediaQuery } from '@mui/material'
+import { ReactNode, useEffect } from 'react'
+import { useSelector } from 'react-redux'
+
+export const ToggleController = ({ children }: { children: ReactNode }) => {
+  const { showSidebar } = useSelector(selectTaskDetails)
+  const matches = useMediaQuery('(max-width:600px)')
+
+  useEffect(() => {
+    if (matches && !matches) {
```

Comment:
Is this condition intentional @aatbip

Reply:
@saugatmaharjan It was a hack but I have added `nonMobile` to avoid confusion in the commit below.